docs/website/docs/HardWonWisdom.md

```diff
@@ -112,3 +112,123 @@ REQ2: If L≤k (see REQ1) and L<R, we must validate at least L+1 of their header
+One might naturally expect this is the stability window (see the above entry in this file for why this is `3k/f` in Praos), but it is instead twice that.
+The best resource we have for motivating this decision is [Edsko de Vries' IOG seminar from 2020 June on the HFC and time](https://drive.google.com/file/d/1QIJ-VBlj-txB6K6E7DIEnY5TzaD89qQm/view).
+
+- At the 16m04s mark he motivates that the forewarning must suffice for at least k+1 headers.
+
+- At the 22m42s mark he starts explaining the Byron proposal state machine, and this ends up motivating "stably stable", aka "doubly stable", aka "double stability", aka "stable stability".
+  (This recording was made before the Byron-to-Shelley fork happened.)
+
+- From 30m46s to 38m52s, he explains the justification for cross-chain translations (and off-handedly that it's technically optional).
+
+Thus the Shelley safe zone is _**two**_ stability windows, as enforced by [the `PPUP` ledger rule](https://github.com/input-output-hk/cardano-ledger/blob/180271602640bcac1214084b6de61d0468332f00/eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Ppup.hs#L192) (that's the tip of the `master` branch at the time of writing this, although this constraint is not new).
+
+TODO In hindsight today, I have some doubts about these conclusions.
+Specifically, I suspect singly-stable could be completely sufficient, at least for ChainSync.
+
+The ultimate answer to the Office Hours question is that the query will fail if and only if its argument is beyond the conservative estimate for the end of the first era without a known end time.
+That estimate will usually be the start of the least epoch that begins more than `6k/f` slots after the tip of ledger state that was used to answer the query.
+
+The key corresponding parts of the implementation are as follows.
+
+- The `ouroboros-consensus:Ouroboros.Consensus.HardFork.History.Qry` module defines `Interpreter`, `Qry`, and `interpretQuery`.
+  Both `Interpeter` and `Qry` are opaque types, so in reality only whatever pre-defined queries that module exports, such as `slotToWallclock`, are available.
+
+- These are ultimately used for every slot/epoch/time translation: `EpochInfo` passed to the ledger, the leadership check in a block-producing node, and all Consesus queries (see `ouroboros-consensus:Ouroboros.Consensus.HardFork.Combinator.Ledger.Query.GetInterpreter`).
+
+- Every downstream use, such as the `query-slot` CLI command ultimately boils down to issuing the `GetInterpreter` query to the node itself and then invoking `interpretQuery`.
+
+- Crucially, that interpretation can fail when the given interpreter cannot answer the given query, which is precisely the topic of the Office Hours question.
+
+- Remark.
+  A significantly fresher `Interpreter` will be able to answer more queries than a stale `Interpreter`.
+  The `ouroboros-consensus:Ouroboros.Consensus.HardFork.History.Caching` module captures that idea.
```

Comment:
Maybe "captures" -> "makes use of", as the most direct way to interpret that remark is to always use the `Interpreter` made from the ledger state of the tip of the current selection.

Also, I am not quite interested in how expensive creating a `Summary`/an `Interpreter` from a `LedgerState` actually is; intuitively, it seems very cheap :thinking: OTOH, this abstraction is super lightweight, so no need to change anything even if it is super cheap.